otherlibs/win32unix/dup.c

```diff
@@ -14,22 +14,104 @@
 /**************************************************************************/
 
 #include <caml/mlvalues.h>
+#include <caml/memory.h>
+#include <caml/fail.h>
 #include "unixsupport.h"
 
-CAMLprim value unix_dup(value cloexec, value fd)
+#define _WIN32_LEAN_AND_MEAN
+#include <winsock2.h>
+
+static HANDLE duplicate_handle(BOOL cloexec, HANDLE oldh)
```

Comment:
```suggestion
static HANDLE duplicate_handle(BOOL inherit, HANDLE oldh)
```
This argument is the negation of `cloexec` and should be named accordingly.